qiskit_nature/operators/second_quantization/sparse_label_op.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import annotations
+from abc import abstractmethod, abstractclassmethod, ABC
+from typing import Dict
+
+from qiskit.opflow.mixins import StarAlgebraMixin
+from qiskit.quantum_info.operators.mixins import TolerancesMixin
+
+class SparseLabelOp(StarAlgebraMixin, TolerancesMixin, ABC):
+    def __init__(self, data: Dict[str, complex], register_length: int = None):
+        self._data = data # stores strings and numbers
+        self._register_length = register_length
+
+    def add(self, other) -> SparseLabelOp:
+        """Return Operator addition of self and other"""
+        new_data = self._data.copy()
+
+        for key, value in other.items():
+            if key in new_data.keys():
+                new_data[key] += value
+            else:
+                new_data[key] = value
+
+        return self.__class__(new_data)
+    
+    def mul(self, other: complex) -> SparseLabelOp:
+        """Return scalar multiplication of self and other"""
+        if not isinstance(other, (int, float, complex)):
+            raise TypeError(
+                f"Unsupported operand type(s) for *: 'SparseLabelOp' and '{type(other).__name__}'"
+            )
+        return self.__class__( #Max to think about...
+            [(label, coeff * other) for label, coeff in self._data],
+            register_length=self.register_length,
+        )
```

Comment:
We had a discussion on this and have decided the following things:
- `SparseLabelOp` should NOT be abstract. This means, you can instantiate it normally here.
- by extension, all methods which were currently planned to be abstract, can be removed. This is not an issue because:
    - `normaled_ordered` is not even well-defined for all planned subclasses
    - `commuting` is no longer required since we skip the implementation of `compose` on this class
    - `zero` and `one` might be implemented directly, now that this class is no longer abstract (but I honestly don't think we really need these here)

The reason we went with this solution, is that subclasses will have to override the mathematical methods anyways, in order to do further compatibility checks. They then internally call `new_sparse_label_op = super().whatever(other)` and will need methods to construct a new instance of their specific subclass, using the data from `new_sparse_label_op`